src/cedict-service/cedict.js

```diff
@@ -0,0 +1,400 @@
+    this._configuration = configuration
+
+    /**
+     * Whether the object is ready to serve data or not.
+     *
+     * @type {boolean}
+     */
+    this.isReady = false
+
+    this._storage = null
+
+    /**
+     * If CEDICT be stored in memory this object will hold all its data.
+     *
+     * @type {{entries: object[]|Map, meta: {}}}
+     */
+    this.cedict = {
+
+      // A dictionary's metadata
+      meta: {},
+
+      metaKey: 1,
+
+      /**
+       * If data is stored in memory `entries` will keep either
+       * an array of dictionary records (if no in memory indexes will be employed) or
+       * a map of dictionary records (if in memory indexes will be used).
+       * If dictionary data will be stored in permanents storage only `entries` will be null
+       */
+      dictionary: null,
+
+      /**
+       * If in memory indexes be used, `traditionalHeadwordsIdx`
+       * will hold a map: Map<traditionalHeadword, Array[entryIndex]>. Otherwise it will be null.
+       */
+      traditionalHeadwordsIdx: null,
+
+      /**
+       * If in memory indexes be used, `simplifiedHeadwordsIdx`
+       * will hold a map: Map<simplifiedHeadword, Array[entryIndex]>. Otherwise it will be null.
+       */
+      simplifiedHeadwordsIdx: null
+    }
+
+    /*
+    If characterForm is not specified in the request we will search for records with the
+    preferred character form specified below first. If any results for that character form will be found,
+    only those ones will be returned to the client. If no results for the preferred character form
+    are in the dictionary then we will search for records with other character forms.
+     */
+    this.preferredCharacterForm = Cedict.characterForms.TRADITIONAL
+
+    /*
+    This is a character form we will fallback into if matches for the preferred one are not found.
+     */
+    this.fallbackCharacterForm = Cedict.characterForms.SIMPLIFIED
+  }
+
+  /**
+   * Checks if the configuration supplied has all the necessary information in it.
+   * If configuration is not valid it will throw an error indicating which check failed.
+   *
+   * @param {object} configuration - A JSON like configuration object.
+   */
+  static checkConfiguration (configuration) {
+    if (!configuration.storage) throw new Error('Storage tree is missing from a configuration')
+    if (!configuration.storage.stores) throw new Error('Stores data is missing from a configuration')
+    if (!configuration.storage.stores.dictionary) throw new Error('Dictionary tree is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex) throw new Error('A primaryIndex tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex.hasOwnProperty('keyPath')) throw new Error('A keyPath option of a primaryIndex tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
```

Comment:
This may be a silly question, but is there a reason you're including `// eslint-disable-line no-prototype-builtins` on all of these lines instead of either writing the code as `Object.prototype.hasOwnProperty.call(configuration.storage.stores.dictionary.primaryIndex, 'keyPath')` or disabling the `no-prototype-builtins` rule in the eslint configuration in `package.json`?

Reply:
We discussed it before and decided that using an `Object.prototype.hasOwnProperty.call` pattern would make code harder to read while not giving any significant advantages as risks that it mitigates are minimal in our case (@balmas please confirm that this still holds). I, however, somehow did not ever think about disabling it in a eslint config. Disabling it once and for all is a great idea, on my opinion, and I will updated package.json to reflect that this rule is off.

FYI: we have a mutual agreement about the other rule, `prefer-const`. We decided to use const only in cases when values of all of its props do not change. In all other cases (when a reference to the object is constant but its props will change) we decided to stick with `let` disabling the eslint rule in that particular case. So:
`const obj = { a:1, b:2 }` means that both `obj` and `a` with `b` will never change
`let obj = { a:1, b:2 }` means that `obj` and/or `a` or `b` will change
`let obj = { a:1, b:2 } // esling-disable-line prefer-const` means that `obj` will remain constant but either `a` or `b` will change
So please don't be surprised seeing constructs like described above in the code.

Reply:
>We discussed it before and decided that using an Object.prototype.hasOwnProperty.call pattern would make code harder to read while not giving any significant advantages as risks that it mitigates are minimal in our case (@balmas please confirm that this still holds).

confirm that this still is true